wordpress-mobile/WordPress-iOS
# Try/media coordinator share context

This PR merges the two Media Coordinator classes MediaUploadCoordinator and MediaSyncCoordinator. 

The reason for this was that we where seeing some duplicate media after uploading a new asset on the library. This happened because the two coordinator use different CoreData context and one was not aware of the other, so when a new Media was created and uploaded, the sync coordinator didn't know anything about it so when it synced he got the new media from the server and duplicated it locally.

The way is implemented now, both process share the same background context and this allow proper sync between both operations.

This PR also reduces some cross dispatching when generating thumbnails that was causing slowdowns when scrolling on the Media Library.

To test:
 - Start the app
 - Open a blog media library
 - Upload some media. 
 - Scroll down to cause a sync while the uploading is ongoing.
 - Check if no duplicate data happens.
